actors/test/cron_catches_expiries_scenario_test.go

```diff
@@ -0,0 +1,160 @@
+	v, err := v.WithEpoch(proveTime)
+	require.NoError(t, err)
+	proveCommitParams := miner.ProveCommitSectorParams{
+		SectorNumber: sectorNumber,
+	}
+	_, code = v.ApplyMessage(worker, minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.ProveCommitSector, &proveCommitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// In the same epoch, trigger cron to validate prove commit
+	_, code = v.ApplyMessage(builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+	require.Equal(t, exitcode.Ok, code)
+
+	// advance to proving period and submit post
+	dlInfo, pIdx, v := vm.AdvanceTillProvingDeadline(t, v, minerAddrs.IDAddress, sectorNumber)
+
+	submitParams := miner.SubmitWindowedPoStParams{
+		Deadline: dlInfo.Index,
+		Partitions: []miner.PoStPartition{{
+			Index:   pIdx,
+			Skipped: bitfield.New(),
+		}},
+		Proofs: []abi.PoStProof{{
+			PoStProof: abi.RegisteredPoStProof_StackedDrgWindow32GiBV1,
+		}},
+		ChainCommitRand: []byte("not really random"),
+	}
+
+	_, code = v.ApplyMessage(addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.SubmitWindowedPoSt, &submitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// add market collateral for client and miner
+	collateral := big.Mul(big.NewInt(3), vm.FIL)
+	_, code = v.ApplyMessage(unverifiedClient, builtin.StorageMarketActorAddr, collateral, builtin.MethodsMarket.AddBalance, &unverifiedClient)
+	require.Equal(t, exitcode.Ok, code)
+	collateral = big.Mul(big.NewInt(64), vm.FIL)
+	_, code = v.ApplyMessage(worker, builtin.StorageMarketActorAddr, collateral, builtin.MethodsMarket.AddBalance, &minerAddrs.IDAddress)
+	require.Equal(t, exitcode.Ok, code)
+
+	// create a deal required by upgrade sector
+	dealIDs := []abi.DealID{}
+	dealStart := v.GetEpoch() + miner.MaxProveCommitDuration[sealProof]
+	deals := publishDeal(t, v, worker, unverifiedClient, minerAddrs.IDAddress, "deal1", 1<<30, false, dealStart, 181*builtin.EpochsInDay)
+	dealIDs = append(dealIDs, deals.IDs...)
+
+	// precommit capacity upgrade sector with deals
+	upgradeSectorNumber := abi.SectorNumber(101)
+	upgradeSealedCid := tutil.MakeCID("101", &miner.SealedCIDPrefix)
+	preCommitParams = miner.SectorPreCommitInfo{
+		SealProof:              sealProof,
+		SectorNumber:           upgradeSectorNumber,
+		SealedCID:              upgradeSealedCid,
+		SealRandEpoch:          v.GetEpoch() - 1,
+		DealIDs:                dealIDs,
+		Expiration:             v.GetEpoch() + 220*builtin.EpochsInDay,
+		ReplaceCapacity:        true,
+		ReplaceSectorDeadline:  dlInfo.Index,
+		ReplaceSectorPartition: pIdx,
+		ReplaceSectorNumber:    sectorNumber,
+	}
+	_, code = v.ApplyMessage(addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.PreCommitSector, &preCommitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// advance time to end of original sector's first proving deadline after minimum prove time has past
+	proveTime = v.GetEpoch() + miner.PreCommitChallengeDelay + 1
```

Comment:
Above you use `proveTime := v.GetEpoch() + miner.MaxProveCommitDuration[sealProof]` which seems right.  Why is this delay between precommit and prove commit different?

Reply:
The first advances to the end of the prove commit time range. That's a little arbitrary. This one advances to the beginning of the prove commit range, then advances till the next proving deadline of the original sector, which should still be in the proving period range. Prove commit of the replacing sector at the last epoch in the original sector's proving deadline is what we need to test.

I'm going to switch the first prove commit to be the minimum and augment the comment for this advancement.